Source/CarthageKit/Project.swift

```diff
@@ -689,50 +710,80 @@ public final class Project {
 	}
 
 	/// Creates symlink between the dependency checkouts and the root checkouts
-	private func symlinkCheckoutPathsForDependencyProject(dependency: ProjectIdentifier, subDependencies: Set<ProjectIdentifier>, rootDirectoryURL: URL) -> SignalProducer<(), CarthageError> {
-		let rootCheckoutsURL = rootDirectoryURL.appendingPathComponent(CarthageProjectCheckoutsPath, isDirectory: true).resolvingSymlinksInPath()
-		let rawDependencyURL = rootDirectoryURL.appendingPathComponent(dependency.relativePath, isDirectory: true)
+	private func symlinkCheckoutPathsForDependencyProject(dependency: Dependency<PinnedVersion>, repositoryURL: URL, rootDirectoryURL: URL, addSubmoduleForDependency: SignalProducer<(), CarthageError>? = nil) -> SignalProducer<(), CarthageError> {
```

Comment:
This should use a Swift 3 name since we're in the process of migrating.

`symlinkCheckoutPaths(for dependency: ...)`